File: java/darts/src/main/java/Darts.java
class Darts {
    private static double OUTER_RADIUS  = 10.0;
    private static double MIDDLE_RADIUS =  5.0;
    private static double INNER_RADIUS  =  1.0;
    private static int OUTER_SCORE  =  1;
    private static int MIDDLE_SCORE =  5;
    private static int INNER_SCORE  = 10;

    int score(double x, double y) {
        double distance = Math.sqrt(x * x + y * y);
        int score = 0;
        if (distance <= INNER_RADIUS) {
            score = INNER_SCORE;
        }
        else if (distance <= MIDDLE_RADIUS) {
            score = MIDDLE_SCORE;
        }
        else if (distance <= OUTER_RADIUS) {
            score = OUTER_SCORE;
        }
        return score;
    }
}

File: euphoria/darts/darts.ex
public function score(atom x, atom y)
    atom dist = sqrt(x * x + y * y)
    if dist <= 1 then
        return 10
    elsif dist <= 5 then
        return 5
    elsif dist <= 10 then
        return 1
    else
        return 0
    end if
end function
